cmd/uplink/cmd/setup.go

```diff
@@ -209,3 +218,37 @@ func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, erro
 	// address is host:
 	return net.JoinHostPort(addressParts[0], defaultPort), nil
 }
+
+// SaveEncryptionKey saves the key in a new file which will be stored in
+// filepath.
+// It returns an error if the directory doesn't exist, the file already exists
+// or there is an I/O error.
+func SaveEncryptionKey(key []byte, filepath string) (err error) {
+	f, err := os.OpenFile(filepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errors.New("directory path doesn't exist")
+		}
+
+		if os.IsExist(err) {
+			return errors.New("file key already exists")
+		}
+
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			err = f.Close()
+		} else {
+			_ = f.Close()
```

Comment:
Because the error returned is relevant over the the one that `Close` may return.

I didn't bear in mind `errs.Combine`, having it solve the issue because you don't miss the important one, but I'm still not sure if when checking such error in logs, I would like to see the error in `Close` when the `Write` happened, because in my opinion will be more noise than something to help

Reply:
Is it possible that `f` is `nil` at this point and `f.Close()` could panic?

Reply:
it cannot because if `os.Open` returns a `nil` error, then `f` cannot be `nil`

Reply:
I still think we should not ignore the error returned by Close. If Write or Chmod fails and Close returns an error too, this additional error may give us additional info for analyzing the problem.

Reply:
Propagate all errors up the chain rather than hiding it. It might contain extra information on why write failed. General rule is, do not hide/ignore the error unless there's a good reason not to (e.g. something else fails as a result).

Reply:
The example doesn't help that much, but don't worry, I see the point considering that we have `errs.Combine`, I wasn't sure until that point we want to add more and more.